client/components/shared/NewBankForm.js

```diff
@@ -32,35 +36,32 @@ export default class NewBankForm extends React.Component {
         return this.refs.password.getDOMNode();
     }
 
-    onChangedBank() {
+    computeState(bank) {
+        assert(typeof bank !== 'undefined', 'There should be at least one bank in the list');
+        return {
+            bankUuid: bank.uuid,
+            customFields: bank.customFields || []
+        };
+    }
+
+    handleChangeBank() {
         let uuid = this.domBank().value;
         let found = store.getStaticBanks().filter(b => (b.uuid === uuid));
-
         assert(found.length === 1, 'selected bank doesnt exist');
         let bank = found[0];
-
-        if (typeof bank.customFields !== 'undefined') {
-            this.setState({
-                hasCustomFields: true,
-                customFields: bank.customFields
-            });
-        } else {
-            this.setState({
-                hasCustomFields: false,
-                customFields: []
-            });
-        }
+        this.setState(this.computeState(bank));
+        this.domBank().focus();
     }
 
-    onSubmit() {
+    handleSubmit() {
         let bank = this.domBank().value;
         let id = this.domId().value.trim();
         let pwd = this.domPassword().value.trim();
         let customFields;
 
-        if (this.state.hasCustomFields) {
+        if (this.state.customFields.length > 0) {
             customFields = this.state.customFields.map((field, index) =>
-                this.refs[`customField${index}`].getValue()
+                this.refs[`customField${index}${this.state.bankUuid}`].getValue()
```

Comment:
It is indeed used to fix a problem I observed. Without using the bankUuid to identify the customField, the defaultValue of the select is not well set.